pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -643,7 +643,8 @@ protected void handleProducerRemoved(Producer producer) {
         }
 
         try {
-            if (!topic.endsWith(EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME)
+            // TODO can all system topics have any type of subscription?
+            if (!TopicName.get(topic).getLocalName().equals(EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME)
```

Comment:
`TRANSACTION_BUFFER_SNAPSHOT` was not included in this check. Should it be?

Reply:
`TRANSACTION_BUFFER_SNAPSHOT` does not need to include this check

Reply:
TRANSACTION_BUFFER_SNAPSHOT does not need to include this check